crypt.go

```diff
@@ -135,6 +139,23 @@ func activate(volumeName, sourceDevicePath string, key []byte, options []string)
 		}
 		done <- true
 	}()
+
+	f, err := os.OpenFile(fifoPath, os.O_WRONLY, 0)
+	if err != nil {
+		// If we fail to open the write end, the read end will be blocked in open()
+		cmd.Process.Kill()
+		return xerrors.Errorf("cannot open FIFO for passing key to systemd-cryptsetup: %w", err)
+	}
+
+	if _, err := f.Write(key); err != nil {
```

Comment:
I wonder if we can have a short write here, resulting in a situation where we have the right key, but it's only partially sent to cryptsetup.

Reply:
This can't happen - the kernel guarantees that writes to pipes of less than one page are atomic - if you write() up to one page to a pipe that's opened with O_NONBLOCK, then either all bytes are written or no bytes are written (Without O_NONBLOCK, the write will block if there isn't enough room to write all of the data)